atmo/clusters/jobs.py

```diff
@@ -32,7 +34,58 @@ def delete_clusters():
 
 
 @newrelic.agent.background_task(group='RQ')
+def update_master_address(cluster_id, force=False):
+    """
+    Update the public IP address for the cluster with the given cluster ID
+    """
+    cluster = Cluster.objects.get(id=cluster_id)
+    # quick way out in case this job was called accidently
+    if cluster.master_address and not force:
+        return
+    # first get the cluster info from AWS
+    info = cluster.get_info()
+    master_address = info.get('public_dns') or ''
```

Comment:
That wouldn't catch an empty string of `public_dns`, which IIRC was an issue in the past